src/test/llm/assisted_decoding_test.cpp

```diff
@@ -0,0 +1,429 @@
+        }
+    )";
+
+    ASSERT_EQ(
+        handler->dispatchToProcessor(endpointCompletions, requestBody, &response, comp, responseComponents, writer),
+        ovms::StatusCode::OK);
+    parsedResponse.Parse(response.c_str());
+    ASSERT_TRUE(parsedResponse["choices"].IsArray());
+    ASSERT_EQ(parsedResponse["choices"].Capacity(), 1);
+    auto& choice = parsedResponse["choices"].GetArray()[0];
+    ASSERT_TRUE(choice["text"].IsString());
+    EXPECT_STREQ(choice["text"].GetString(), expectedMessages[0].c_str());
+
+    // Dynamic number of candidates
+    requestBody = R"(
+        {
+            "model": "lm_cb_speculative",
+            "stream": false,
+            "temperature": 0,
+            "max_tokens": 10,
+            "prompt": "What is OpenVINO?",
+            "assistant_confidence_threshold": 0.4
+        }
+    )";
+
+    ASSERT_EQ(
+        handler->dispatchToProcessor(endpointCompletions, requestBody, &response, comp, responseComponents, writer),
+        ovms::StatusCode::OK);
+    parsedResponse.Parse(response.c_str());
+    ASSERT_TRUE(parsedResponse["choices"].IsArray());
+    ASSERT_EQ(parsedResponse["choices"].Capacity(), 1);
+    choice = parsedResponse["choices"].GetArray()[0];
+    ASSERT_TRUE(choice["text"].IsString());
+    EXPECT_STREQ(choice["text"].GetString(), expectedMessages[0].c_str());
+}
+
+TEST_F(AssistedDecodingPipelinesHttpTest, unaryChatCompletionsJsonSpeculativeDecoding) {
+    GTEST_SKIP() << "Skip this test until strange tokenizer behavior is fixed";
+    // Generate reference from the base model (unassisted generation)
+    config.max_new_tokens = 10;
+    config.temperature = 0;
+    ASSERT_EQ(generateExpectedText("What is OpenVINO?", false), 0);
+    ASSERT_EQ(config.num_return_sequences, expectedMessages.size());
+
+    // Static number of candidates
+    std::string requestBody = R"(
+        {
+            "model": "lm_cb_speculative",
+            "stream": false,
+            "temperature": 0,
+            "max_tokens": 10,
+            "num_assistant_tokens": 3,
+            "messages": [
+            {
+                "role": "user",
+                "content": "What is OpenVINO?"
+            }
+            ]
+        }
+    )";
+
+    ASSERT_EQ(
+        handler->dispatchToProcessor(endpointChatCompletions, requestBody, &response, comp, responseComponents, writer),
+        ovms::StatusCode::OK);
+    parsedResponse.Parse(response.c_str());
+    ASSERT_TRUE(parsedResponse["choices"].IsArray());
+    ASSERT_EQ(parsedResponse["choices"].Capacity(), 1);
+    auto& choice = parsedResponse["choices"].GetArray()[0];
+    ASSERT_TRUE(choice["message"].IsObject());
+    ASSERT_TRUE(choice["message"]["content"].IsString());
+    ASSERT_TRUE(choice["finish_reason"].IsString());
+    ASSERT_FALSE(choice["logprobs"].IsObject());
+    ASSERT_EQ(choice["message"]["content"].GetString(), expectedMessages[0]);
+
+    // Dynamic number of candidates
+    /*
+    requestBody = R"(
```

Comment:
why is it in comment, but not another test?

Reply:
debugging leftover, I'll uncomment it
note that this test is skipped anyway for now